internal/generationtest/generation_test.go

```diff
@@ -0,0 +1,77 @@
+package generationtest
+
+import (
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/bazel-gazelle/testtools"
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+	yaml "gopkg.in/yaml.v2"
+)
+
```

Comment:
Then in this file you'd do something like (after adding an import for `flag`) to add in some flags:

```
var (
  gazelleBinaryPath = flag.String("gazelle_binary_path", "", "Path to the gazelle binary to test")
  testData = flag.String("test_data", "", "Path to the data you want to test")
)
```

Now you can use those strings to get the values. I believe that `testData` is going to be some kind of delimited string, like `"foo.txt,bar.txt"` or maybe space delimited, I'm not sure, but you should be able to use [`strings.Split`](https://pkg.go.dev/strings#Split) to turn it into a list of input files. I think you can iterate through that list of files looking for a `WORKSPACE` and use that as the root for execution of Gazelle.

[0] https://blog.jbowen.dev/2019/08/using-go-flags-in-tests/